### ISSUE TITLE
Order of elements in data

### ISSUE THREAD

**Issue by cwittern** [Jinntec/Fore#63]
Hi, this looks very promising, and I am at the moment just playing around to evaluate its usefulness for my projects.
I looked at the TEI header demo and played around, for example removing and adding an <author>.  The resulting <data> on the same demopage now show the following:
`<TEI>
    <teiHeader>
        <fileDesc>
            <titleStmt>
                <title>Main title for the electronic version of the text</title>
                <title type="sub">One or more additional titles, which may have a @type</title>
                <author>Rudi Rüssel</author>
                
                <editor>Heinz Hund</editor>
            <author>Christian Wittern</author></titleStmt>
            <publicationStmt>
                <publisher>Name of the institution publishing the electronic version</publisher>
                <date when="2021-09-24"/>
                <availability>
                    <licence target="http://creativecommons.org/licenses/by/3.0/">Creative Commons Attribution
                        ShareAlike 3.0 License
                    </licence>
                </availability>
            </publicationStmt>
            <sourceDesc>
                <p>A free text paragraph describing the manuscript used</p>
            </sourceDesc>
        </fileDesc>
        <profileDesc>
            <langUsage>
                <language ident="de"/>
            <language ident="eng"/></langUsage>
            <textClass>
                <keywords>
                    <term>XML</term>
                    <term>TEI</term>
                </keywords>
            </textClass>
        </profileDesc>
        <revisionDesc>
            <change when="2021-05-31T09:04:25+02:00">teiHeader was automatically produced from the database</change>
        </revisionDesc>
    </teiHeader>
    <text>
        <body>
            <p>Some text here.</p>
        </body>
    </text>
</TEI>
`
The new author is appended to the end of the element, while the visual editing has the editors after the authors.  My question is:  Is this a bug or a feature? Can this be adjusted?
I would expect the order of the elements be the same in both cases.
All the best,
Chris

**Comment by JoernT** [comment]
you're right Chris. This is still a glitch in the implementation. The demo uses the older `fx-append` action which is not accurate. There no if `fx-insert` action which allows much more flexibility and should handle this case. I leave this ticket open til i've re-written the example to use `fx-insert`

**Comment by cwittern** [comment]
Ah, I see, thanks for pointing this out.  But this means that the code itself allows for this case, which is good news.  Keep up the good work, this looks very promising and has the potential to solve a lot of headaches for me:-)

**Comment by JoernT** [comment]
will need to add a couple of insert tests.
